### DISCUSSION TITLE
Getting RangeError when using faker.string.alphanumeric without specifying max

### DISCUSSION THREAD

**Discussion by skrishnan22** [faker-js/faker#2372] [Q&A] [answered]
I am using **faker.string.alphanumeric with only min length** specified in options. It gives a rangeError. Works fine if both min and max are specified.

`faker.string.alphanumeric({ length: { min: 10 } })`

Is this expected behaviour i.e length option should have both min and max specified always?

If not, I am assuming this could be because the Number.MAX_SAFE_INTEGER used in rangeToNumber [here](https://github.com/faker-js/faker/blob/651d1a8cd11b2d562c7a6473b8a544b81d5fbb95/src/modules/string/index.ts#L321) is greater than max Array length.

Complete error trace here

````txt
RangeError: Invalid array length
    at Function.from (<anonymous>)
    at A.multiple (<workspace>/node_modules/@faker-js/faker/dist/cjs/modules/helpers/index.js:3:723)
    at x.fromCharacters (<workspace>/node_modules/@faker-js/faker/dist/cjs/modules/string/index.js:1:1049)
    at x.alphanumeric (<workspace>/node_modules/@faker-js/faker/dist/cjs/modules/string/index.js:1:1901)
    at Object.<anonymous> (<workspace>/test-faker.js:3:26)
    at Module._compile (node:internal/modules/cjs/loader:1254:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1308:10)
    at Module.load (node:internal/modules/cjs/loader:1117:32)
    at Module._load (node:internal/modules/cjs/loader:958:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
````

**Comment by ST-DDT** [comment, marked as answer]
According to our API docs: `number | { min: number, max: number }` the length.max parameter is required.

https://fakerjs.dev/api/string.html#alphanumeric

**Reply by matthewmayer** [reply]
Should it throw with a more useful error if one is provided without the other?

**Reply by skrishnan22** [reply]
Got it. Thanks!. I guess the confusion originated because `faker.string.alphanumeric({ length: { max: 10 } })`
works. So assumed passing only min would also work.

**Reply by ST-DDT** [reply]
The parameter is required and if you dont use typescript, there are a lot of other things that you could do wrong as well.
IMO it is not worth checking for these errors explicitly, we have the compiler for that already.

**Reply by matthewmayer** [reply]
But the TS compiler allows you to provide max only. Only min throws an error:
```
faker.string.alphanumeric({ length: { max: 10 } }) //works
faker.string.alphanumeric({ length: { min: 10 } }) //throws an error
```

**Reply by ST-DDT** [reply]
Why would the compiler do that? Min is also required!

https://github.com/faker-js/faker/blob/76886ad96d694268856de83747d7c754b6504050/src/modules/string/index.ts#L276-L292

Note: I'm talking about the dev+compile time TypeScript compiler + IDE tooling. Not the actual invocation at runtime.

EDIT: I tested it myself with VSCode and the compiler does mark it as error:

![VSCode error](https://github.com/faker-js/faker/assets/1579362/3586c024-2825-4e87-97cf-2408d65afb51)

**Reply by matthewmayer** [reply]
Ok I was just testing it with REPL.
